src/main/java/org/reactome/addlinks/fileprocessors/RGDFileProcessor.java

```diff
@@ -0,0 +1,64 @@
+package org.reactome.addlinks.fileprocessors;
+
+import org.reactome.addlinks.EnsemblBiomartUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RGDFileProcessor extends FileProcessor{
+
+    public RGDFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public RGDFileProcessor()
+    {
+        super();
+    }
+
+    private static final int rgdIdentifierIndex = 0;
+    private static final int uniprotIdentifiersIndex = 21;
+
+    /**
+     * Build map of UniProt identifiers to Rat Genome Database (RGD) identifiers that is used to create RGD cross-references in database.
+     * @return - Map<String, List<String>
+     */
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFilePath = Paths.get(this.pathToFile.toAbsolutePath().toString());
```

Comment:
I'm a bit confused about this line.  `this.pathToFile` is already a Path object - could it be used instead of creating `inputFilePath`?

Reply:
Yep good catch, for cases where the file isn't gzipped toy can just use `this.pathToFile`